sycl/doc/extensions/experimental/sycl_ext_oneapi_annotated_ptr.asciidoc

```diff
@@ -417,6 +421,8 @@ annotated_ptr operator+(size_t Offset) const;
 |
 Returns an `annotated_ptr` that points to `this[Offset]`.
 
+Only applicable properties will be passed to the returned `annotated_ptr`.
+
```

Comment:
is each property supposed to document whether it is propagated through operator+(size_t)?
> Yes I think so

is alignment propagated? This would make sense depending on the value of offset
>alignment is propagated to annotated_ref, but it will not be propagated in operator[]. As you said it's indeed depending on the value of offset. So for some operators we have to drop the alignment properties.

Reply:
No it can't depend on the value. The value is only known at runtime but the property is propagated at compile time.

Reply:
I think this change also is not a good idea.  It might make sense to propagate the properties through the pointer arithmetic operations, depending on the (runtime) value of the offset.  I think it should be the user's responsibility to drop the property explicitly when necessary.  For example, the description of `alignment` already says that it is the user's responsibility to ensure that the pointer has the indicated alignment:

> The behavior is undefined if the pointer value does not have the indicated alignment.

We should not design the API in a way that tries to guess what the user wants.  Instead, we should have simple rules, and let the user explicitly drop properties when necessary.

Reply:
I think if we ask users to do this then this could end up being a common mistake? If user applying alignment mistakenly, the error varies depending on the user code. It could be a memory optimization issue that are vague to be debugged or it could be something else. I think we should try to avoid this as much as we can